src/test/java/mate/academy/springbootintro/service/ShoppingCartServiceTest.java

```diff
@@ -0,0 +1,219 @@
+    import mate.academy.springbootintro.model.CartItem;
+    import mate.academy.springbootintro.model.ShoppingCart;
+    import mate.academy.springbootintro.model.User;
+    import mate.academy.springbootintro.repository.book.BookRepository;
+    import mate.academy.springbootintro.repository.cartitem.CartItemRepository;
+    import mate.academy.springbootintro.repository.shoppingcart.ShoppingCartRepository;
+    import org.junit.jupiter.api.DisplayName;
+    import org.junit.jupiter.api.Test;
+    import org.junit.jupiter.api.extension.ExtendWith;
+    import org.mockito.InjectMocks;
+    import org.mockito.Mock;
+    import org.mockito.Mockito;
+    import org.mockito.junit.jupiter.MockitoExtension;
+    import java.util.Optional;
+    import java.util.Set;
+
+    @ExtendWith(MockitoExtension.class)
+    public class ShoppingCartServiceTest {
+
+        private static final Long USER_ID = 1L;
+        private static final Long BOOK_ID = 10L;
+        private static final Long SHOPPING_CART_ID = 1L;
+        private static final Long CART_ITEM_ID = 10L;
+        private static final int QUANTITY = 5;
+        private static final String BOOK_TITLE = "Test Book";
+
+        @InjectMocks
+        private ShoppingCartServiceImpl shoppingCartService;
+
+        @Mock
+        private ShoppingCartRepository shoppingCartRepository;
+
+        @Mock
+        private ShoppingCartMapper shoppingCartMapper;
+
+        @Mock
+        private BookRepository bookRepository;
+
+        @Mock
+        private CartItemRepository cartItemRepository;
+
+        @Test
+        @DisplayName("""
+                Verify addToCart() method works
+                """)
+        public void addToCart_ValidRequestDto_ReturnsShoppingCartDto() {
+            // Given
+            AddBookToCartRequest request = new AddBookToCartRequest(
+                    BOOK_ID,
+                    QUANTITY
+            );
+
+            ShoppingCart shoppingCart = new ShoppingCart();
+            shoppingCart.setId(SHOPPING_CART_ID);
+
+            User user = new User();
+            user.setId(USER_ID);
+            shoppingCart.setUser(user);
+
+            Book book = new Book();
+            book.setId(BOOK_ID);
+            book.setTitle(BOOK_TITLE);
+
+            CartItem cartItem = new CartItem();
+            cartItem.setBook(book);
+            cartItem.setShoppingCart(shoppingCart);
+            cartItem.setQuantity(QUANTITY);
+
+            CartItemDto cartItemDto = new CartItemDto(
+                    cartItem.getId(),
+                    cartItem.getBook().getId(),
+                    cartItem.getBook().getTitle(),
+                    cartItem.getQuantity()
+            );
+
+            Set<CartItemDto> expectedCartItems = Set.of(cartItemDto);
+
+            ShoppingCartDto expected = new ShoppingCartDto(
+                    shoppingCart.getId(),
+                    shoppingCart.getUser().getId(),
+                    expectedCartItems
+            );

```

Comment:
You can create private methods to create those test objects, then use these methods here to make tests more readable.

Reply:
and then you can re-use these private methods in all your test methods:)